File: sources/academy/webscraping/scraping_basics_python/exercises/unesco_links.py
import httpx
from bs4 import BeautifulSoup
from urllib.parse import urljoin

listing_url = "https://www.unesco.org/en/countries"
response = httpx.get(listing_url)
response.raise_for_status()

soup = BeautifulSoup(response.text, "html.parser")

for country in soup.select(".node--type-country"):
    link = country.select_one("a")
    if link and 'href' in link.attrs:
        url = urljoin(listing_url, link["href"])
        print(url)

File: sources/academy/webscraping/scraping_basics_javascript/exercises/unesco_links.mjs
import * as cheerio from 'cheerio';

const listingUrl = 'https://www.unesco.org/en/countries';
const response = await fetch(listingUrl);

if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

const html = await response.text();
const $ = cheerio.load(html);

for (const element of $('.node--type-country').toArray()) {
  const $link = $(element).find('a').first();
  const url = new URL($link.attr('href'), listingUrl).href;
  console.log(url);
}
